includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -93,6 +94,20 @@ public function __construct(
 			$this->id           = self::GATEWAY_ID . '_' . $this->stripe_id;
 			$this->method_title = "WooCommerce Payments ($this->title)";
 		}
+
+		add_filter( 'woocommerce_available_payment_gateways', [ $this, 'add_sepa_gateway' ], 100, 1 );
+	}
+
+	/**
+	 * Adds SEPA gateway
+	 *
+	 * @param array $gateways                  - Filtered gateways.
+	 */
+	public function add_sepa_gateway( $gateways ) {
+		if ( WC_Payments_Features::is_upe_split_enabled() && 'sepa_debit' === $this->stripe_id ) {
```

Comment:
We probably need to extend this `if` so that it checks for the subsccription cart only, because otherwise the sepa gateway might be added twice (in case of the non subscription product)